### PR TITLE
New 'Repos' page and consolidation of 'Apps' page

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Alongside the refactoring in #3509, there are some related visual improvements. Here is a summary of the changes:

## Addition of a 'Repos' page

#3481 saw the addition of a 'repos.json' alongside the longstanding 'apps.json' functionality. So it seems only natural that we should also want a summary page of Repos alongside our summary page of Apps.

repos page screenshot

## Addition of 'Repos' link in header

See above.

## Changes to 'Apps' page

The addition of the 'Repos' page led to some duplication etc, which got consolidated and visually reviewed, and eventually led to the following improvements to the Apps page. It removed the need for a separate 'by team' page.

Before:

before

After:

after

## Removal of the 'apps by team' page

The [existing page](https://docs.publishing.service.gov.uk/apps/by-team.html) is no longer required once these changes are merged.

Trello: https://trello.com/c/oUsgzZnk/247-standardise-how-we-represent-repositories-in-the-dev-docs

### CODE REVIEW

File: source/apps.html.md.erb [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+<% end %>
+
+## Apps by type
+
+Apps can be loosely grouped into different types:
+
+- **APIs** provide functionality to other apps. For example, an app can send an email via Email Alert API.
+- **Frontend apps** render content to GOV.UK users. For example, a [HMRC manual page][hmrc-manual] is rendered by an application called [manuals-frontend][manuals-frontend]. Use the [chrome extension][extension] to find out which application
+is rendering any given page, and [read more about the frontend architecture on GOV.UK](/manual/frontend-architecture.html).
+- **Publishing apps** are used by editors to publish content to GOV.UK. For example, [specialist-publisher][specialist-publisher] publishes [specialist documents][specialist documents]. The apps are secured by behind a [single signon system][signon]. They use an [omniauth adapter called gds-sso][gds-sso] to authenticate the user.
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply source/apps.html.md.erb:25]: Agreed!

**Comment by ChrisBAshton** [comment]
Thanks for the review, @benthorner! I've addressed all the individual comments if you'd like to re-review 👀 

> Do we really need the "apps" page anymore? All the info that's on it can be seen on the new "repos" page.

Part of the reason to properly support the concept of 'repos' rather than just 'apps' was the confusion some teams were facing when looking at what used to be `applications.yaml`, and seeing non-apps in there. We wanted to include non-apps so that we could surface all relevant documentation via search, but it led to confusion with stakeholders, particularly around things like Rails upgrades. Think "we've only updated 7 apps, but I see we're responsible for 10, why haven't we done the other 3 yet?", when the other 3 are gems etc.

So for _that_ reason, I think it's still worth having a distinct grouping of 'apps' vs 'repos'.

> Seeing both of them, I half expected the links on the "apps" page to go to the live app, but what we really mean by "app" is "app repository", not the running version of it.

Yes, that's a good point. I think once this is merged we could look at making the app and app-overview pages a bit more tailored to apps. Don't think it's a blocker for this PR though.

> Looking at the code itself, do you think it's worth adding some tests for the new / updated pages? There's [some precedent for it](https://github.com/alphagov/govuk-developer-docs/tree/main/spec/pages) but I also get that this isn't _that_ much of a change from what was there before.

The existing tests are really just linting: [redirect_spec.rb](https://github.com/alphagov/govuk-developer-docs/blob/main/spec/pages/redirects_spec.rb) checks that redirects aren't missing their `.html` suffix, and [manual_page_spec.rb](https://github.com/alphagov/govuk-developer-docs/blob/main/spec/pages/manual_page_spec.rb) checks that every page has an owner, a title, etc. I don't think we really have a precedent for testing the contents of pages.
More tests in general would probably be a good thing, but I agree that the changes in this PR aren't very different to what was there before and so doesn't introduce any new risks.

File: source/apps.html.md.erb [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+
+| Type | Count | Applications |
+| --- | --- | --- |
+<% Repos.active_apps.group_by(&:team).sort.each do |name, apps| %>
+| # <%= name %> | <%= apps.count %> | <%= apps.map { |app| "[#{app.repo_name}](/repos/#{app.repo_name}.html)" }.join(", ") %> |
+<% end %>
+
+## Apps by type
+
+Apps can be loosely grouped into different types:
```
[the remark this answers is not in the document]
Comment (benthorner) [reply source/apps.html.md.erb]: I'm assuming you weren't keen on this suggestion ;-).

**Review by benthorner — APPROVED** [review]
Ready to 🚢.

I like the argument of keeping the page with a view to tailoring it in future. And in the meantime the mysterious "stakeholders" can still benefit from it 😜.

File: source/apps.html.md.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply source/apps.html.md.erb]: No, I implemented your suggestion! https://github.com/alphagov/govuk-developer-docs/pull/3510/commits/7d91f98d2d020dd9b68f78660fd7e10f8039e2d5

I did make a mistake with the original rebase, but it should be there now 🤔

**Pull request merged**